server/anchor.py

```diff
@@ -175,52 +171,46 @@ class NotReadyException(AnchorException):
-        self._aml_config_path = AML_CONFIG
-        self._did = None
-        self._init_error = None
-        self._pool = None
+        self._anonymous = not LEDGER_SEED
+        self._cache: LedgerCache = None
+        self._aml_config_path: str = AML_CONFIG
+        self._did: str = None
+        self._init_error: str = None
+        self._pool: indy_vdr.Pool = None
         self._protocol = protocol or DEFAULT_PROTOCOL
         self._ready = False
-        self._ledger_lock = None
-        self._register_dids = REGISTER_NEW_DIDS and not ANONYMOUS
-        self._seed = seed_as_bytes(LEDGER_SEED)
-        self._sync_lock = None
+        self._ledger_lock: asyncio.Lock = None
+        self._register_dids = bool(REGISTER_NEW_DIDS and LEDGER_SEED)
+        self._seed = seed_as_bytes(LEDGER_SEED) if LEDGER_SEED else None
+        self._sync_lock: asyncio.Lock = None
         self._syncing = False
-        self._taa_accept = None
+        self._taa_accept: str = None
         self._taa_config_path = TAA_CONFIG
-        self._verkey = None
+        self._verkey: str = None
 
     async def _open_pool(self):
-        pool_name = "nodepool"
-        pool_cfg = {}
         self._pool = None
+        attempts = 0
 
-        try:
-            await pool.set_protocol_version(self._protocol)
-        except IndyError as e:
-            raise AnchorException("Error setting pool protocol version") from e
+        while True:
+            try:
+                genesis = await resolve_genesis_file()
 
-        # remove existing pool config by the same name in ledger browser mode
-        try:
-            pool_names = {cfg["pool"] for cfg in await pool.list_pools()}
-            if pool_name in pool_names:
-                await pool.delete_pool_ledger_config(pool_name)
-        except IndyError as e:
-            raise AnchorException("Error deleting existing pool configuration") from e
+                LOGGER.info("Connecting to ledger pool")
 
-        try:
-            await pool.create_pool_ledger_config(
-                pool_name, json.dumps({"genesis_txn": await resolve_genesis_file()})
-            )
-        except IndyError as e:
-            raise AnchorException("Error creating pool configuration") from e
+                indy_vdr.set_protocol_version(self._protocol)
+                self._pool = await open_pool(transactions_path=genesis)
 
-        try:
-            self._pool = await pool.open_pool_ledger(pool_name, json.dumps(pool_cfg))
-        except IndyError as e:
-            raise AnchorException("Error opening pool ledger connection") from e
+                LOGGER.info("Finished pool refresh: %s", self._pool.last_status)
+            except VdrError as e:
+                if e.code == VdrErrorCode.POOL_TIMEOUT and attempts < 5:
+                    LOGGER.info("Pool timeout occurred, waiting to retry")
+                    attempts += 1
+                    await asyncio.sleep(10)
+                    continue
```

Comment:
I like this!  Please make the number of attempts and delay between attempts configurable.

Reply:
Done.